falcon/responders.py

```diff
@@ -18,11 +18,6 @@
 from falcon.status_codes import HTTP_405
 
 
-def path_not_found(req, resp, **kwargs):
-    """Simply sets responseto "404 Not Found", no body."""
-    resp.status = HTTP_404
```

Comment:
I would prefer keeping this function and modifying it to raise falcon.HTTPNotFound(). That way, we can avoid any unintended side-effects; `API.__call__()` is currently written in way that assumes `API._get_responder()` does not raise an error.